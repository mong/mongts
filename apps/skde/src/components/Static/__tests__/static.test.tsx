/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { PageContent } from "..";

type frontMatterType = {
  title: string;
  lang: "nb" | "en" | "nn";
  ingress?: string;
};

test("BokmÃ¥l render", async () => {
  const frontMatter: frontMatterType = {
    title: "Dette er en tittel",
    lang: "nb",
    ingress: "Dette er et sammendrag",
  };
  const markdownText = "Dette er en kort tekst";
  const { container } = render(
    <PageContent frontMatter={frontMatter} content={markdownText} />,
  );
  expect(container).toMatchSnapshot();
});

test("Nynorsk render", async () => {
  const frontMatter: frontMatterType = {
    title: "Dette er ein tittel",
    lang: "nn",
    ingress: "Dette er eit samandrag",
  };
  const markdownText = "Dette er ein kort tekst";
  const { container } = render(
    <PageContent frontMatter={frontMatter} content={markdownText} />,
  );
  expect(container).toMatchSnapshot();
});

test("English render", async () => {
  const frontMatter: frontMatterType = {
    title: "This is the title",
    lang: "en",
    ingress: "This is a summary",
  };
  const markdownText =
    "Dette er en test\n\n# Hovedoverskrift\nLitt mer tekst\n## Underoverskrift\n### Dette er en overskrift!\n\n```sas\ndata test;\nset test2;\nvar = 0;\nrun;\n```\n\nOg litt tekst her";
  const { container } = render(
    <PageContent frontMatter={frontMatter} content={markdownText} />,
  );
  expect(container).toMatchSnapshot();
});
