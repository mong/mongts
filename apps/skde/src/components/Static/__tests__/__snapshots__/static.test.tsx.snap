// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bokmål render 1`] = `
<div>
  <header
    class="_headerContainer_d17b29"
  >
    <div
      class="_headerLogo_d17b29"
    >
      <a
        href="/helseatlas"
        rel="home"
      >
        <img
          alt="Hjem"
          data-nimg="1"
          decoding="async"
          height="47"
          loading="lazy"
          src="/helseatlas/img/logos/helseatlas.svg?w=384&q=75"
          srcset="/helseatlas/img/logos/helseatlas.svg?w=256&q=75 1x, /helseatlas/img/logos/helseatlas.svg?w=384&q=75 2x"
          style="color: transparent;"
          width="186"
        />
      </a>
    </div>
    <nav
      class="_headerNavContainer_d17b29"
    >
      <div
        class="_navButtons_d17b29"
      >
        <div
          class="_lang_d17b29"
        >
          <div
            class="_no_d17b29 _active_d17b29"
            data-testid="buttonNo"
          >
            <a
              href="/helseatlas"
            >
              NO
            </a>
          </div>
          <div
            class="_eng_d17b29 undefined"
            data-testid="buttonEng"
          >
            <a
              href="/helseatlas/en"
            >
              ENG
            </a>
          </div>
        </div>
        <div
          class="_menuButton_d17b29"
        >
          <div>
            <div
              class="_popUpBtn_d1cb5c"
            >
              <button
                class="_button_4123ea"
                data-testid="menuButton"
              >
                <span
                  class="_buttonBurgerIcon_4123ea"
                >
                  <span
                    class="top"
                  />
                  <span
                    class="middle"
                  />
                  <span
                    class="bottom"
                  />
                </span>
                <span
                  class="_buttonText_4123ea"
                >
                  MENY
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="_container_6b07e8"
  >
    <div
      class="_article__title_6b07e8"
    >
      <h1>
        Dette er en tittel
      </h1>
    </div>
    <div
      class="_article_6b07e8"
    >
      <div
        class="_article__content_6b07e8"
      >
        <div
          class="_ingress_e666a6"
        >
          <p>
            Dette er et sammendrag
          </p>
        </div>
        <p>
          Dette er en kort tekst
        </p>
      </div>
    </div>
  </div>
  <footer
    class="_footerContainer_2cbb36"
  >
    <div
      class="_footerContent_2cbb36"
    >
      <div
        class="_footerTop_2cbb36"
      >
        <strong>
          Senter for klinisk dokumentasjon og evaluering (SKDE)
        </strong>
        <br />
        Postboks 6
        <br />
        9038 Tromsø
        <br />
        <br />
        <strong>
          Helse Førde
        </strong>
        <br />
        Postboks 1000
        <br />
        6807 Førde
      </div>
      <div
        class="_footerBottom_2cbb36"
      >
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-nord.no/"
            title="Link til Helse Nord"
          >
            <img
              alt="Helse Nord logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-nord-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://www.skde.no/"
            title="Link til SKDE"
          >
            <img
              alt="SKDE logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/skde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/skde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/skde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-forde.no/"
            title="Link til Helse Førde"
          >
            <img
              alt="Helse Førde logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-forde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-vest.no/"
            title="Link til Helse Vest"
          >
            <img
              alt="Helse Vest logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-vest-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`English render 1`] = `
<div>
  <header
    class="_headerContainer_d17b29"
  >
    <div
      class="_headerLogo_d17b29"
    >
      <a
        href="/helseatlas/en"
        rel="home"
      >
        <img
          alt="Hjem"
          data-nimg="1"
          decoding="async"
          height="47"
          loading="lazy"
          src="/helseatlas/img/logos/helseatlas.svg?w=384&q=75"
          srcset="/helseatlas/img/logos/helseatlas.svg?w=256&q=75 1x, /helseatlas/img/logos/helseatlas.svg?w=384&q=75 2x"
          style="color: transparent;"
          width="186"
        />
      </a>
    </div>
    <nav
      class="_headerNavContainer_d17b29"
    >
      <div
        class="_navButtons_d17b29"
      >
        <div
          class="_lang_d17b29"
        >
          <div
            class="_no_d17b29 undefined"
            data-testid="buttonNo"
          >
            <a
              href="/helseatlas"
            >
              NO
            </a>
          </div>
          <div
            class="_eng_d17b29 _active_d17b29"
            data-testid="buttonEng"
          >
            <a
              href="/helseatlas/en"
            >
              ENG
            </a>
          </div>
        </div>
        <div
          class="_menuButton_d17b29"
        >
          <div>
            <div
              class="_popUpBtn_d1cb5c"
            >
              <button
                class="_button_4123ea"
                data-testid="menuButton"
              >
                <span
                  class="_buttonBurgerIcon_4123ea"
                >
                  <span
                    class="top"
                  />
                  <span
                    class="middle"
                  />
                  <span
                    class="bottom"
                  />
                </span>
                <span
                  class="_buttonText_4123ea"
                >
                  MENU
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="_container_6b07e8"
  >
    <div
      class="_article__title_6b07e8"
    >
      <h1>
        This is the title
      </h1>
    </div>
    <div
      class="_article_6b07e8"
    >
      <div
        class="_article__content_6b07e8"
      >
        <div
          class="_ingress_e666a6"
        >
          <p>
            This is a summary
          </p>
        </div>
        <p>
          Dette er en test
        </p>
        

        <h1
          id="hovedoverskrift"
        >
          Hovedoverskrift
        </h1>
        

        <p>
          Litt mer tekst
        </p>
        

        <h2
          id="underoverskrift"
        >
          Underoverskrift
        </h2>
        

        <h3
          id="dette-er-en-overskrift"
        >
          Dette er en overskrift!
        </h3>
        

        <pre>
          <code
            class="language-sas"
          >
            data test;
set test2;
var = 0;
run;

          </code>
        </pre>
        

        <p>
          Og litt tekst her
        </p>
      </div>
    </div>
  </div>
  <footer
    class="_footerContainer_2cbb36"
  >
    <div
      class="_footerContent_2cbb36"
    >
      <div
        class="_footerTop_2cbb36"
      >
        <strong>
          Senter for klinisk dokumentasjon og evaluering (SKDE)
        </strong>
        <br />
        Postboks 6
        <br />
        9038 Tromsø
        <br />
        <br />
        <strong>
          Helse Førde
        </strong>
        <br />
        Postboks 1000
        <br />
        6807 Førde
      </div>
      <div
        class="_footerBottom_2cbb36"
      >
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-nord.no/"
            title="Link til Helse Nord"
          >
            <img
              alt="Helse Nord logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-nord-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://www.skde.no/"
            title="Link til SKDE"
          >
            <img
              alt="SKDE logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/skde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/skde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/skde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-forde.no/"
            title="Link til Helse Førde"
          >
            <img
              alt="Helse Førde logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-forde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-vest.no/"
            title="Link til Helse Vest"
          >
            <img
              alt="Helse Vest logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-vest-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Nynorsk render 1`] = `
<div>
  <header
    class="_headerContainer_d17b29"
  >
    <div
      class="_headerLogo_d17b29"
    >
      <a
        href="/helseatlas"
        rel="home"
      >
        <img
          alt="Hjem"
          data-nimg="1"
          decoding="async"
          height="47"
          loading="lazy"
          src="/helseatlas/img/logos/helseatlas.svg?w=384&q=75"
          srcset="/helseatlas/img/logos/helseatlas.svg?w=256&q=75 1x, /helseatlas/img/logos/helseatlas.svg?w=384&q=75 2x"
          style="color: transparent;"
          width="186"
        />
      </a>
    </div>
    <nav
      class="_headerNavContainer_d17b29"
    >
      <div
        class="_navButtons_d17b29"
      >
        <div
          class="_lang_d17b29"
        >
          <div
            class="_no_d17b29 _active_d17b29"
            data-testid="buttonNo"
          >
            <a
              href="/helseatlas"
            >
              NO
            </a>
          </div>
          <div
            class="_eng_d17b29 undefined"
            data-testid="buttonEng"
          >
            <a
              href="/helseatlas/en"
            >
              ENG
            </a>
          </div>
        </div>
        <div
          class="_menuButton_d17b29"
        >
          <div>
            <div
              class="_popUpBtn_d1cb5c"
            >
              <button
                class="_button_4123ea"
                data-testid="menuButton"
              >
                <span
                  class="_buttonBurgerIcon_4123ea"
                >
                  <span
                    class="top"
                  />
                  <span
                    class="middle"
                  />
                  <span
                    class="bottom"
                  />
                </span>
                <span
                  class="_buttonText_4123ea"
                >
                  MENY
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="_container_6b07e8"
  >
    <div
      class="_article__title_6b07e8"
    >
      <h1>
        Dette er ein tittel
      </h1>
    </div>
    <div
      class="_article_6b07e8"
    >
      <div
        class="_article__content_6b07e8"
      >
        <div
          class="_ingress_e666a6"
        >
          <p>
            Dette er eit samandrag
          </p>
        </div>
        <p>
          Dette er ein kort tekst
        </p>
      </div>
    </div>
  </div>
  <footer
    class="_footerContainer_2cbb36"
  >
    <div
      class="_footerContent_2cbb36"
    >
      <div
        class="_footerTop_2cbb36"
      >
        <strong>
          Senter for klinisk dokumentasjon og evaluering (SKDE)
        </strong>
        <br />
        Postboks 6
        <br />
        9038 Tromsø
        <br />
        <br />
        <strong>
          Helse Førde
        </strong>
        <br />
        Postboks 1000
        <br />
        6807 Førde
      </div>
      <div
        class="_footerBottom_2cbb36"
      >
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-nord.no/"
            title="Link til Helse Nord"
          >
            <img
              alt="Helse Nord logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-nord-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-nord-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://www.skde.no/"
            title="Link til SKDE"
          >
            <img
              alt="SKDE logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/skde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/skde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/skde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-forde.no/"
            title="Link til Helse Førde"
          >
            <img
              alt="Helse Førde logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-forde-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-forde-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
        <div
          class="_logo_2cbb36"
        >
          <a
            href="https://helse-vest.no/"
            title="Link til Helse Vest"
          >
            <img
              alt="Helse Vest logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75"
              srcset="/helseatlas/img/logos/helse-vest-hvit.svg?w=256&q=75 1x, /helseatlas/img/logos/helse-vest-hvit.svg?w=384&q=75 2x"
              style="color: transparent;"
              width="180"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
